src/server/game/Guilds/Guild.cpp

```diff
@@ -2400,8 +2400,8 @@ void Guild::DeleteMember(SQLTransaction& trans, ObjectGuid guid, bool isDisbandi
     // Call script on remove before member is actually removed from guild (and database)
     sScriptMgr->OnGuildRemoveMember(this, player, isDisbanding, isKicked);
 
-    if (Member* member = GetMember(guid))
```

Comment:
To prevent double lookup (first in GetMember, then in .erase), refactor as
```
auto memberItr = m_members.find(guid);
if (memberItr != m_members.end())
{
    delete memberItr->second;
    m_members.erase(memberItr);
}
```

Reply:
I do see your point regarding auto here, QtCreator (without ClangCodeModel because it newer works) could not autocomplete memberItr. I know recent Visual Studio's have no issues whatsoever without auto though.

Reply:
There is another, bigger, problem with `auto` that seems to always be ignored by everyone. Your IDE is not the only place where you read code, github pull requests is another place and you have none of the tools available to determine variable type

Reply:
I don't see the problem there though considering you generally don't write code using GH and we have CI to say if members of a class variable exists or not.